core/node/provider.go

```diff
@@ -114,10 +119,35 @@ https://github.com/ipfs/kubo/blob/master/docs/config.md#routingaccelerateddhtcli
 					return false
 				}, sampledBatchSize))
 		}
-		sys, err := provider.New(repo.Datastore(), opts...)
+
+		var strategyChanged bool
+		ctx := context.Background()
+		ds := repo.Datastore()
+		strategyKey := datastore.NewKey(reprovideStrategyKey)
+
+		prev, err := ds.Get(ctx, strategyKey)
+		if err != nil && !errors.Is(err, datastore.ErrNotFound) {
+			logger.Error("cannot read previous reprovide stratedy", "err", err)
+		} else if string(prev) != reprovideStrategy {
+			strategyChanged = true
+		}
+
+		sys, err := provider.New(ds, opts...)
 		if err != nil {
 			return nil, err
 		}
+		if strategyChanged {
+			logger.Infow("reporvide strategy changed, clearing provide queue", "previous", string(prev), "current", reprovideStrategy)
```

Comment:
nit: use config key name
```suggestion
			logger.Infow("Reprovider.Strategy changed, clearing provide queue", "previous", string(prev), "current", reprovideStrategy)
```